lib/core/sessions.js

```diff
@@ -625,17 +625,24 @@ class ServerSessionPool {
 
 // TODO: this should be codified in command construction
 // @see https://github.com/mongodb/specifications/blob/master/source/read-write-concern/read-write-concern.rst#read-concern
-function commandSupportsReadConcern(command) {
-  return (
+function commandSupportsReadConcern(command, options) {
+  if (
     command.aggregate ||
     command.count ||
     command.distinct ||
     command.find ||
-    command.mapReduce ||
     command.parallelCollectionScan ||
     command.geoNear ||
     command.geoSearch
-  );
+  ) {
+    return true;
+  }
+
+  if (command.mapReduce && (options.out && options.out.inline !== 1 && options.out !== 'inline')) {
```

Comment:
I think this is backwards. According to the spec, `readConcern` is available for "`mapReduce` command where the `out` option is `{ inline: 1 }`". This seems to be checking that `out` is **not** `inline`.

Reply:
oops, nice catch again :) fixed.